### DISCUSSION TITLE
No Sudo switch for binary installs does not work

### DISCUSSION THREAD

**Discussion by goreha-abhilash** [gruntwork-io/knowledge-base#745] [Help] [answered]
Hello

I am working on a non admin mac device. IT Admins installed the gruntwork-installer for me via Admin account. However, I still cannot install binaries with no-sudo switch as mentioned in the official documentation for binary installation. For reference, I am trying to install this:
gruntwork-install --binary-name "kubergrunt" --repo "https://github.com/gruntwork-io/kubergrunt" --tag "v0.11.3"  --no-sudo

The admins also tried to run this via admin account but the no-sudo switch does not give any output.

---

<ins datetime="2023-06-29T12:06:03Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110309">Tracked in ticket #110309</a></p>
</ins>

**Comment by pete0emerson** [comment, marked as answer]
This runs a bit counter-intuitive to what I would expect (and I think you are expecting), but `--no-sudo` is not a binary flag, it is expecting a value. Try this:

```
 gruntwork-install --binary-name "kubergrunt" --repo "https://github.com/gruntwork-io/kubergrunt" --tag "v0.11.3" --no-sudo true
```

Here's my run, first without specifying `true` and then with:

```
 ~ $ gruntwork-install --binary-name "kubergrunt" --repo "https://github.com/gruntwork-io/kubergrunt" --tag "v0.11.3" --no-sudo
 ~ $ gruntwork-install --binary-name "kubergrunt" --repo "https://github.com/gruntwork-io/kubergrunt" --tag "v0.11.3" --no-sudo true
2023-06-29 09:38:25 [INFO] [gruntwork-install] Installing kubergrunt...
[fetch] INFO[2023-06-29T09:38:26-04:00] Downloading release asset kubergrunt_darwin_amd64 to /tmp/gruntwork-script-modules/kubergrunt_darwin_amd64
[fetch] INFO[2023-06-29T09:38:29-04:00] Downloaded /tmp/gruntwork-script-modules/kubergrunt_darwin_amd64
[fetch] INFO[2023-06-29T09:38:29-04:00] Download of release assets complete
2023-06-29 09:38:29 [INFO] [gruntwork-install] Moving /tmp/gruntwork-script-modules/kubergrunt_darwin_amd64 to /usr/local/bin/kubergrunt and setting execute permissions
2023-06-29 09:38:29 [INFO] [gruntwork-install] Success!
```

**Comment by goreha-abhilash** [comment]
Thanks Pete for this clarification. Tried this out but now a strange permission denied error:
gruntwork-install --binary-name "kubergrunt" --repo "https://github.com/gruntwork-io/kubergrunt" --tag "v0.11.3" --no-sudo true 
2023-06-29 16:07:19 [INFO] [gruntwork-install] Installing kubergrunt...
[fetch] INFO[2023-06-29T16:07:19+02:00] Downloading release asset kubergrunt_darwin_amd64 to /tmp/gruntwork-script-modules/kubergrunt_darwin_amd64 
[fetch] INFO[2023-06-29T16:07:22+02:00] Downloaded /tmp/gruntwork-script-modules/kubergrunt_darwin_amd64 
[fetch] INFO[2023-06-29T16:07:22+02:00] Download of release assets complete          
2023-06-29 16:07:22 [INFO] [gruntwork-install] Moving /tmp/gruntwork-script-modules/kubergrunt_darwin_amd64 to /usr/local/bin/kubergrunt and setting execute permissions
mv: rename /tmp/gruntwork-script-modules/kubergrunt_darwin_amd64 to /usr/local/bin/kubergrunt: Permission denied

Are you running this in a standard user in a Mac?

**Reply by pete0emerson** [reply]
I am running this as a standard user on my Mac, yes. I have run `chmod 0775 pete:admin /usr/local/bin`. I think I did that initially for Homebrew, although I'm not positive.

You can also specify an alternate binary installation directory for `gruntwork-install` with ` --binary-install-dir`.

**Reply by goreha-abhilash** [reply]
Thanks Pete. Along with that I also had to update the permissions for /etc/user-data folder to get the binaries to install. 
But this does not seem to work for modules. Do you have any ideas/suggestions for modules installation as well?

**Reply by pete0emerson** [reply]
I can check. What module(s) are you trying to install, and when you try to install them, do they give you a similar failed `mv` command?

**Reply by goreha-abhilash** [reply]
Here is the command output along with the module name I am trying to install:

$ gruntwork-install --repo "https://github.com/gruntwork-io/terraform-aws-ci" \                          
                                          --module-name "infrastructure-deploy-script" \
                                          --tag "$MODULE_CI_VERSION" --no-sudo true
2023-07-05 16:13:19 [INFO] [gruntwork-install] Repository is not public. GITHUB_OAUTH_TOKEN environment variable is required.
2023-07-05 16:13:19 [INFO] [gruntwork-install] Installing from infrastructure-deploy-script...
2023-07-05 16:13:19 [INFO] [gruntwork-install] Downloading module infrastructure-deploy-script from https://github.com/gruntwork-io/terraform-aws-ci
INFO[2023-07-05T16:13:20+02:00] Downloading tag "v0.51.3" of https://github.com/gruntwork-io/terraform-aws-ci ...  binary=fetch version=
INFO[2023-07-05T16:13:21+02:00] Extracting files from <repo>/modules/infrastructure-deploy-script to /tmp/gruntwork-script-modules/infrastructure-deploy-script ...  binary=fetch version=
INFO[2023-07-05T16:13:21+02:00] 25 files extracted                            binary=fetch version=
INFO[2023-07-05T16:13:21+02:00] Download and file extraction complete.        binary=fetch version=
2023-07-05 16:13:21 [INFO] [gruntwork-install] Setting GRUNTWORK_INSTALL_TAG to v0.51.3
2023-07-05 16:13:21 [INFO] [gruntwork-install] Setting GRUNTWORK_INSTALL_BRANCH to 
2023-07-05 16:13:21 [INFO] [gruntwork-install] Executing /tmp/gruntwork-script-modules/infrastructure-deploy-script/install.sh 
Password:
Abhilash is not in the sudoers file.  This incident will be reported.

**Comment by pete0emerson** [comment]
Per the `--help`, `--no-sudo` only applies to binaries.

```
  --no-sudo		When true, don't use sudo to install the binary into the install directory. Only applies to binaries (not modules). Default: false
```

**Reply by goreha-abhilash** [reply]
Ok, so I am kind of stuck here. Wonder if the admin installs it for me creates a problem
